controllers/rhmi/rhmi_controller.go

```diff
@@ -1419,3 +1441,63 @@ func isInstallationOlderThan1Minute(installation *rhmiv1alpha1.RHMI) bool {
 	aMinuteAgo := time.Now().Add(-time.Minute)
 	return aMinuteAgo.After(installation.CreationTimestamp.Time)
 }
+
+func validateAddOnStsRoleArnParamaterPattern(client k8sclient.Client, namespace string) (bool, error) {
+	// function is checking if STS addon parameter Pattern is valid
+	// Parameter is Valid only in case:
+	// 1.	Parameter exists and value matching AWS Role ARN pattern
+	// Parameter is Not valid  in other cases:
+	// 2.	parameter exists and value is NOT matching AWS Role ARN pattern
+	// 3.	parameter exists and value is empty
+	// 4.	parameter does not exists
+
+	awsArnPattern := "arn:aws:iam:\\S*:\\d+:role\\/\\S+"
+	r, err := regexp.Compile(awsArnPattern)
+	if err != nil {
+		log.Error("regexp Compile error: ", err)
+		return false, err
+	}
+
+	stsRoleArn, stsFound, err := addon.GetStringParameterByInstallType(
+		context.TODO(),
+		client,
+		rhmiv1alpha1.InstallationTypeManagedApi,
+		namespace,
+		addonStsArnParameterName,
+	)
+	if err != nil {
+		log.Error("failed while retrieving addon parameter", err)
+		return false, err
+	} else if stsFound && r.MatchString(stsRoleArn) { //case 1 - VALID
+		log.Info("ARN pattern is valid")
+		return true, nil
+	} else if stsFound && len(stsRoleArn) == 0 { //case 2 - NOT valid
+		log.Info("AWS STS role ARN parameter validation failed - parameter value is empty")
+		return false, nil
+	} else if !stsFound { //case 3 - NOT valid
+		log.Info("AWS STS role ARN parameter validation failed - parameter not found")
+		return false, nil
+	} else if stsFound && !r.MatchString(stsRoleArn) { //case 4 - NOT valid
```

Comment:
Minor: this can be just `} else {` I believe.

Reply:
Oh, I now I see there is `else` block below.
What kind of other, "unknown", issue could there be? 🤔

Reply:
I just removed word "unknown" : ), all remains same, as logic seems to me is ok. The order of cases in code could be any actually. But if you think that it's not so correct, maybe we could talk and I will update. Thank you.